pyhealth/tasks/mortality_prediction.py

```diff
@@ -184,6 +184,71 @@ def mortality_prediction_eicu_fn(patient: Patient):
     return samples
 
 
+def mortality_prediction_eicu_fn2(patient: Patient):
+    """Processes a single patient for the mortality prediction task.
+
+    Mortality prediction aims at predicting whether the patient will decease in the
+    next hospital visit based on the clinical information from current visit
+    (e.g., conditions and procedures).
+
+    Similar to mortality_prediction_eicu_fn, but with different code mapping:
+    - using admissionDx and diagnosisString table as condition codes
+    - using treatment table as procedure codes
+
+    Args:
+        patient: a Patient object
+
+    Returns:
+        samples: a list of samples, each sample is a dict with patient_id,
+            visit_id, and other task-specific attributes as key
+
+    Note that we define the task as a binary classification task.
+
+    Examples:
+        >>> from pyhealth.datasets import eICUDataset
+        >>> eicu_base = eICUDataset(
+        ...     root="/srv/local/data/physionet.org/files/eicu-crd/2.0",
+        ...     tables=["admissionDx", "diagnosisString", "treatment"],
+        ...     code_mapping={},
+        ...     dev=True
+        ... )
+        >>> from pyhealth.tasks import mortality_prediction_eicu_fn2
+        >>> eicu_sample = eicu_base.set_task(mortality_prediction_eicu_fn2)
+        >>> eicu_sample.samples[0]
+        [{'visit_id': '130744', 'patient_id': '103', 'conditions': [['42', '109', '98', '663', '58', '51']], 'procedures': [['1']], 'label': 0}]
+    """
+    samples = []
+    # we will drop the last visit
+    for i in range(len(patient) - 1):
+        visit: Visit = patient[i]
+        next_visit: Visit = patient[i + 1]
+
+        if next_visit.discharge_status not in ["Alive", "Expired"]:
+            mortality_label = 0
+        else:
+            mortality_label = 0 if next_visit.discharge_status == "Alive" else 1
+
+        admissionDx = visit.get_code_list(table="admissionDx")
+        diagnosisString = visit.get_code_list(table="diagnosisString")
+        treatment = visit.get_code_list(table="treatment")
+
+        # exclude: visits without treatment, admissionDx, diagnosisString
+        if len(admissionDx) * len(diagnosisString) * len(treatment) == 0:
+            continue
+        # TODO: should also exclude visit with age < 18
+        samples.append(
+            {
+                "visit_id": visit.visit_id,
+                "patient_id": patient.patient_id,
+                "conditions": [admissionDx] + [diagnosisString],
+                "procedures": [treatment],
+                "label": mortality_label,
+            }
```

Comment:
I would suggest merging this mortality2 func with the original mortality. The final samples can have more keys "conditions", "procedures", "drugs", "admDX+diagString", "treatment", ... Later, user can choose a subset as their feature keys and choose drugs or treatment as their target key.

Reply:
It's feasible, just that it will drop a lot of data as it only keeps data containing both entries: 
```python
if len(conditions) * len(procedures) * len(drugs) == 0:
        continue
```
I am afraid it will break some of the previous pipeline/model performance, so I created another task instead. Another option is to pass a feature key set, like ```key_set = {'condition: ['admissionDx', 'diagnosis], 'procedures' = ['treatment', 'medical']}```, to indicate the features we want to keep. But that would change the unified format of the tasks class.

Reply:
You are right. Let us keep two functions then. 

Could you please help add minor descriptions below each "def readmission_prediction_eicu_fn" to differentiate two functions?